src/common/main.lua

```diff
@@ -45,7 +57,25 @@ main.init = function()
 	utils.createActionHandler(ActionMap.StateNotice, function(msg)
 		local ant = utils.parseAntState(msg.Data)
 		-- we pass in the reference as the state nonce
-		utils.updateAffiliations(msg.From, ant, ADDRESSES, ANTS, tonumber(msg.Reference))
+		local updateResult = utils.updateAffiliations(msg.From, ant, ADDRESSES, ANTS, tonumber(msg.Reference))
+		local affiliatesToRemoveAntIdFrom = updateResult.affiliatesToRemoveAntIdFrom
+
+		-- this may need to be batched for operation within limits of hyperbeam messages, specifically http header size
+		local aclMap = utils.affiliationsForAnt(msg.From, ANTS)
+
+		for _, affiliate in ipairs(affiliatesToRemoveAntIdFrom) do
+			aclMap[affiliate] = utils.affiliationsForAddress(affiliate, ANTS)
+			-- TODO: DELETE the ant from the affiliations on the address mapping. Currently do not know how to do this, or if its even possible with the current state of patch@1.0\
+			-- this will set the mapping to { Owned: [], Controlled: [] } in the hb state
+
+			-- aclMap[affiliate].Owned[utils.indexOf(aclMap[affiliate].Owned, msg.From)] = nil
+			-- aclMap[affiliate].Controlled[utils.indexOf(aclMap[affiliate].Controlled, msg.From)] = nil
+		end
+
+		ao.send({
+			device = "patch@1.0",
+			cache = { acl = aclMap },
```

Comment:
We should consider tests around this in the future surrounding our cache management and how we want to update/delete keys in hyperbeam

Reply:
Both in the current integration setup and ideally we run a local hyperbeam node so we can test its interactions